muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/api/api.ts

```diff
@@ -0,0 +1,275 @@
+import {
+  AnnouncerApi,
+  CeeposApi,
+  CommunicatorApi,
+  Configuration,
+  CoursepickerApi,
+  CredentialsApi,
+  DiscussionApi,
+  EvaluationApi,
+  FeedsApi,
+  GuiderApi,
+  HopsApi,
+  HopsUpperSecondaryApi,
+  MaterialsApi,
+  MatriculationApi,
+  MeApi,
+  NotesApi,
+  OrganizationApi,
+  PedagogyApi,
+  RecordsApi,
+  UserApi,
+  UsergroupApi,
+  UserpluginApi,
+  WorklistApi,
+  WorkspaceApi,
+  WorkspaceDiscussionApi,
+  WorkspaceNotesApi,
+} from "../generated/client";
+
+const isDevelopment = process.env.NODE_ENV !== "production";
+
+const configuration = new Configuration({
+  basePath: isDevelopment
+    ? window.location.origin
+    : "https://otavanopisto.muikkuverkko.fi",
```

Comment:
Hard-coded domain is not going to work with installations on different domains (duh).